src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SemanticFacts/CSharpSemanticFacts.cs

```diff
@@ -57,23 +57,54 @@ public bool CanReplaceWithRValue(SemanticModel semanticModel, [NotNullWhen(true)
         {
             var location = token.GetLocation();
 
-            foreach (var ancestor in token.GetAncestors<SyntaxNode>())
+            var tokenParent = token.Parent;
+            var sourceAncestor = tokenParent;
```

Comment:
i don't see the need for both of these locals.  it looks like you don't use tokenParent past this point.  So just do `var soureAncestor = token.Parent;`, and then also update sourceAncestor below if necessary.